gradio/components/multimodal_textbox.py

```diff
@@ -0,0 +1,196 @@
+            autofocus: If True, will focus on the textbox when the page loads. Use this carefully, as it can cause usability issues for sighted and non-sighted users.
+            elem_id: An optional string that is assigned as the id of this component in the HTML DOM. Can be used for targeting CSS styles.
+            elem_classes: An optional list of strings that are assigned as the classes of this component in the HTML DOM. Can be used for targeting CSS styles.
+            render: If False, component will not render be rendered in the Blocks context. Should be used if the intention is to assign event listeners now but render the component later.
+            text_align: How to align the text in the textbox, can be: "left", "right", or None (default). If None, the alignment is left if `rtl` is False, or right if `rtl` is True. Can only be changed if `type` is "text".
+            rtl: If True and `type` is "text", sets the direction of the text to right-to-left (cursor appears on the left of the text). Default is False, which renders cursor on the right.
+            autoscroll: If True, will automatically scroll to the bottom of the textbox when the value changes, unless the user scrolls up. If False, will not scroll to the bottom of the textbox when the value changes.
+            show_submit_button: If False, will not show the submit button. Only applies if `interactive` is True.
+        """
+        self.file_types = file_types
+        if value is None:
+            value = {"text": "", "files": []}
+        if file_types is not None and not isinstance(file_types, list):
+            raise ValueError(
+                f"Parameter file_types must be a list. Received {file_types.__class__.__name__}"
+            )
+        self.lines = lines
+        self.max_lines = max(lines, max_lines)
+        self.placeholder = placeholder
+        self.show_submit_button = show_submit_button
+        self.autofocus = autofocus
+        self.autoscroll = autoscroll
+
+        super().__init__(
+            label=label,
+            info=info,
+            every=every,
+            show_label=show_label,
+            container=container,
+            scale=scale,
+            min_width=min_width,
+            interactive=interactive,
+            visible=visible,
+            elem_id=elem_id,
+            elem_classes=elem_classes,
+            render=render,
+            value=value,
+        )
+        self.rtl = rtl
+        self.text_align = text_align
+
+    def preprocess(
+        self, payload: dict[str, str | list] | None
+    ) -> dict[str, str | list] | None:
+        """
+        Parameters:
+            payload: the text and list of file(s) entered in the multimodal textbox.
+        Returns:
+            Passes text value and list of file(s) as a {dict} into the function.
+        """
+        return None if payload is None else payload
+
+    def _download_files(self, value: str | list[str]) -> str | list[str]:
```

Comment:
I don't think we need `_download_files` here. We added this to `File` to get the size but that's not used in the multimodal textbox.

https://github.com/gradio-app/gradio/pull/7620/files#r1515022822